pkg/operator/ceph/controller/spec.go

```diff
@@ -758,25 +759,28 @@ func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Contai
 }
 
 // CreateExternalMetricsEndpoints creates external metric endpoint
-func createExternalMetricsEndpoints(namespace string, monitoringSpec cephv1.MonitoringSpec, ownerInfo *k8sutil.OwnerInfo) (*v1.Endpoints, error) {
+func createExternalMetricsEndpoints(namespace string, monitoringSpec cephv1.MonitoringSpec, ownerInfo *k8sutil.OwnerInfo) (*discoveryv1.EndpointSlice, error) {
 	labels := AppLabels("rook-ceph-mgr", namespace)
+	endpointPortName := ServiceExternalMetricName
+	endpointPortPort := int32(monitoringSpec.ExternalMgrPrometheusPort)
+	endpointPortProtocol := v1.ProtocolTCP
 
-	endpoints := &v1.Endpoints{
+	endpoints := &discoveryv1.EndpointSlice{
```

Comment:
Can you test by running locally,

The update scenario and see the endpoint is successfully using the new endpoint type, maybe by printing it

Reply:
Also, make sure the min k8s version supports it, currently, we are using 1.22.17 ig

Reply:
> Also, make sure the min k8s version supports it, currently, we are using 1.22.17 ig

The API was introduced in 1.17 and became stable in 1.21, so I think we are good in that aspect.

> Can you test by running locally,
> The update scenario and see the endpoint is successfully using the new endpoint type, maybe by printing it

Sure, will update here once the testing is done